contrib/log_correlation/log4j/src/main/java/io/opencensus/contrib/logcorrelation/log4j/OpenCensusTraceContextDataInjector.java

```diff
@@ -0,0 +1,219 @@
+      "io.opencensus.contrib.logcorrelation.log4j."
+          + "OpenCensusTraceContextDataInjector.spanSelection";
+
+  private final Tracer tracer = Tracing.getTracer();
+
+  private final SpanSelection spanSelection;
+
+  /**
+   * How to decide whether to add tracing data from the current span to a log entry.
+   *
+   * @since 0.16
+   */
+  public enum SpanSelection {
+
+    /**
+     * Never add tracing data to log entries. This constant disables the log correlation feature.
+     *
+     * @since 0.16
+     */
+    NO_SPANS,
+
+    /**
+     * Add tracing data to a log entry iff the current span is sampled.
+     *
+     * @since 0.16
+     */
+    SAMPLED_SPANS,
+
+    /**
+     * Always add tracing data to log entries, even when the current span is not sampled. This is
+     * the default.
+     *
+     * @since 0.16
+     */
+    ALL_SPANS
+  }
+
+  /**
+   * Returns the {@code SpanSelection} setting for this instance.
+   *
+   * @return the {@code SpanSelection} setting for this instance.
+   * @since 0.16
+   */
+  public SpanSelection getSpanSelection() {
+    return spanSelection;
+  }
+
+  /**
+   * Constructor to be called by Log4j.
+   *
+   * <p>This constructor looks up the {@link SpanSelection} using the system property {@link
+   * #SPAN_SELECTION_PROPERTY_NAME}.
+   *
+   * @since 0.16
+   */
+  public OpenCensusTraceContextDataInjector() {
+    this(lookUpSpanSelectionProperty());
+  }
+
+  private OpenCensusTraceContextDataInjector(SpanSelection spanSelection) {
+    this.spanSelection = spanSelection;
+  }
+
+  private static SpanSelection lookUpSpanSelectionProperty() {
+    String spanSelectionProperty = System.getProperty(SPAN_SELECTION_PROPERTY_NAME);
+    return spanSelectionProperty == null || spanSelectionProperty.isEmpty()
+        ? DEFAULT_SPAN_SELECTION
+        : parseSpanSelection(spanSelectionProperty);
+  }
+
+  private static SpanSelection parseSpanSelection(String spanSelection) {
+    try {
+      return SpanSelection.valueOf(spanSelection);
+    } catch (IllegalArgumentException e) {
+      return DEFAULT_SPAN_SELECTION;
+    }
+  }
+
+  // The implementation of this method is based on the example in the Javadocs for
+  // ContextDataInjector.injectContextData.
+  @Override
+  public StringMap injectContextData(@Nullable List<Property> properties, StringMap reusable) {
+    if (properties == null || properties.isEmpty()) {
+      return rawContextData();
+    }
+    // Context data has precedence over configuration properties.
+    ThreadContextDataInjector.copyProperties(properties, reusable);
+    reusable.putAll(rawContextData());
+    return reusable;
+  }
+
+  // This method avoids getting the current span when the feature is disabled, for efficiency.
+  @Override
+  public StringMap rawContextData() {
+    switch (spanSelection) {
+      case NO_SPANS:
+        return getContextData();
+      case SAMPLED_SPANS:
+        SpanContext spanContext = tracer.getCurrentSpan().getContext();
```

Comment:
one thing to highlight about this approach is that it requires static initialization (which is not a big deal as much of census does, but to highlight).. This implies you can't do this unless you have a singleton which is the same instance for the entire lifecycle.

I know it isn't designed this way, but this is one reason why having a "context only" api can be handy as you can initialize that way ahead of usage by things like this, and multiple instances of a tracer can use the same context store.